### PR TITLE
Skip UnexpectedRootClaim failures

### PR DESCRIPTION
**Pull request by maurelian**
Five test suites are skipped on the forked upgrade test path. 

They all fail with "UnexpectedRootClaim" errors, so assistance will be needed from the proofs team to ensure they are properly restored.

### CODE REVIEW

**Comment by maurelian** [comment]
* **#13428** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/13428?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/13428?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#13424** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/13424?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#13323** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/13323?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#13375](https://github.com/ethereum-optimism/optimism/pull/13375) <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/13375?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#13471](https://github.com/ethereum-optimism/optimism/pull/13471) <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/13471?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by Inphi** [comment]
These tests are failing likely at least because the dispute game tests assume the `AnchorStateRegistry` is empty, so they create games that claim a low L2 block number.

There's this check in the FDG that asserts the most recent L2 block number anchor precedes the L2 block number that's claimed by dispute games - https://github.com/ethereum-optimism/optimism/blob/0d0f69d59cfbce954710636efdf9812d9a38fead/packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol#L304.

I gather that the upgrade script loads live contracts from the superchain-registry. So the anchored block number read from the op-mainnet `AnchorStateRegistry` at the fork block state will be much higher than the L2 block number used to create dispute games in the test. Hence the revert on L304.

We can fix this problem by tweaking the tests to use inspect the `AnchorStateRegistry` to determine the L2 block numbers to use for game creation.

**Comment by maurelian** [comment]
@Inphi that's helpful thank you!

Unfortunately, while it does help me get past that error in `setUp()`, it reveals a much wider range of issues that will need to be debugged. For now I think it's best to stick to the plan to revisit these tests in https://github.com/ethereum-optimism/optimism/issues/13411

File: packages/contracts-bedrock/justfile [hunk 1]
```diff
@@ -68,6 +68,7 @@ print-pinned-block-number:
 #   Reusing the default block number greatly speeds up the test execution time by caching the
 #   rpc call responses in ~/.foundry/cache/rpc. The default block will need to be updated
 #   when the L1 chain is upgraded.
+# TODO(opcm upgrades): unskip the "NMC" tests which fail on the forked upgrade path with "UnexpectedRootClaim" errors.
```
Comment (blmalone) [inline packages/contracts-bedrock/justfile:71]: What does NMC stand for?

**Review by blmalone — APPROVED** [review]

File: packages/contracts-bedrock/justfile (hunk 1 above)
Comment (maurelian) [reply packages/contracts-bedrock/justfile:71]: --no-match-contract, I can clarify this tho

**Pull request merged**